department-of-veterans-affairs/caseflow
# Ryan appeals/25267 Receipt Date Filtering

Resolves [Add Filtering to Receipt Date](https://jira.devops.va.gov/browse/APPEALS-25267)

# Description
Added receipt date filtering to the Document table. Includes validation and correctly filters.

## Acceptance Criteria

## Testing Plan
All documents in the table are set to august 8th for their date, therefore we will work around that date to test each of the possible filters.
1)Log in as BVAAABshire
2)Navigate to the reader page in caseflow
3)Inside to documents table, you will notice a filter icon next to the 'Reciept Date' column. Click it.
4) Click the dropdown that is titled 'select...' and change it to 'between these dates' 
5) fill in a date before august 8 2023, and after august 8 2023 in their respective columns. The dates may also be on august 
     8th, but both cannot be at the same time. Hit filter, and the table will show the documents still. 
6) Change one of the dates so that august 8th is no longer included in the date range. Once filtered it will remove the 
     documents.
7) Click the dropdown and switch to 'before this date', pick any date that is after or on august 8th and the documents will 
    remain. any date prior will cause the documents to disappear.
8) Click the dropdown and switch to 'after this date'. pick any date that is before or on august 8th and the documents will 
    remain. any date further will cause the documents to disappear. 
9) Click the dropdown and switch to 'on this date'. pick august 8th and the documents will remain. Any other date will cause 
    the documents to dissapear.



# Frontend
## User Facing Changes

 BEFORE|AFTER
<img width="1323" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/856a14bf-9153-4fc7-956f-146fe2716f5d">|
<img width="1088" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/c9b0a312-e9c0-4882-af65-b86dbf8c6b68">

### Error Handling
